filebeat/crawler/prospector.go

```diff
@@ -98,18 +103,43 @@ func (p *Prospector) Run(wg *sync.WaitGroup) {
 }
 
 func (p *Prospector) Stop() {
+	logp.Debug("prospector", "Stopping %d harvesters.", len(p.harvesters))
+	for _, h := range p.harvesters {
+		go h.Stop()
+	}
+	logp.Debug("prospector", "Waiting for %d harvesters to stop", len(p.harvesters))
+	p.harvestersWaitGroup.Wait()
```

Comment:
how is p.done used? the prospector loop starting new harvesters must be stopped before shutting down harvesters